File: math/valid_boomerang.rb
# @param {Integer[][]} points
# @return {Boolean}
def is_boomerang(points)
    x1, y1 = points[0]
    x2, y2 = points[0]
    x3, y3 = points[0]

    ((y2 - y1) * (x3 - x2)) != ((y3 - y2) * (x2 - x1))
end

File: math/valid_boomerang.py
class Solution:
    def isBoomerang(self, points: List[List[int]]) -> bool:
        p1, p2, p3 = points[0], points[1], points[2]
        
        cross_product = (p2[0] - p1[0]) * (p3[1] - p1[1]) - (p2[1] - p1[1]) * (p3[0] - p1[0])
        
        return cross_product != 0